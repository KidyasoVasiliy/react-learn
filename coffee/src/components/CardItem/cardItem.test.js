import React from 'react';
import { CardItem } from './index';
import { shallow } from 'enzyme';

describe('Testing <CardItem />', () => {
  it('CardItem have renderer correctly', () => {
    const cardItem = shallow(<CardItem />);
    expect(cardItem).toMatchSnapshot();
  })
});
