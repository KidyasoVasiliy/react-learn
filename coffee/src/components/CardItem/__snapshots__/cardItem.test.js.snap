// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <CardItem /> CardItem have renderer correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardItem />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          className="cardItem__img"
          src="http://via.placeholder.com/150x150"
        />,
        <div
          className="cardItem__title"
        />,
        undefined,
        <div
          className="cardItem__price"
        />,
      ],
      "className": "cardItem default",
      "onClick": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": undefined,
          "className": "cardItem__img",
          "src": "http://via.placeholder.com/150x150",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "cardItem__title",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "cardItem__price",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            className="cardItem__img"
            src="http://via.placeholder.com/150x150"
          />,
          <div
            className="cardItem__title"
          />,
          undefined,
          <div
            className="cardItem__price"
          />,
        ],
        "className": "cardItem default",
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": undefined,
            "className": "cardItem__img",
            "src": "http://via.placeholder.com/150x150",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "cardItem__title",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "cardItem__price",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
